### DISCUSSION TITLE
Help/question with Delete all orphaned entities option

### DISCUSSION THREAD

**Discussion by L0bit0** [frenck/spook#814] [Q&A]
Hi

I am starting to use spook to clean my HA. 

I launched "Delete all orphaned entities" action/service but after this i still can see a long list of orphaned entities.

I expected to do not have these entities anymore, I am missing anything? or spook jut remove the data of these entities, but entities are still on the DB?

I am missing anything?

Some additional info.

- I also create a script based on the documentation:

![image](https://github.com/user-attachments/assets/c4e6bec3-264f-45be-839c-64784f72cc38)

I restarted HA before check again...

Can you help me to understand if there is an issue or i am missing anything?

Thanks!

**Comment by frenck** [comment]
You are mixing up cleaning up of orhpaned entities in the state machine and registries vs the database history. Those are not the same things.

**Reply by L0bit0** [reply]
Thanks for your quick response.

Sorry, but I am not following you, what "Delete all orphaned entities" should do? I am assuming this should delete history of all orphan entities, so once executed this, "List all orphaned database entities" should not return any entries, correct?

I now realize the script will remove data history on the DB, but not the entity itself.

Sorry in advance if I am missing any concept.

Thanks!
